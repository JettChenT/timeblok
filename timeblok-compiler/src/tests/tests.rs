use crate::{compile_deterministic, ir::ExactDateTime};
use insta::{assert_snapshot, glob};
use std::fs;

macro_rules! assert_compile_snapshot {
    ($input:expr, $base_time:expr) => {
        assert_snapshot!(compile_deterministic($input, $base_time.clone()).unwrap());
    };

    ($input:expr) => {
        assert_compile_snapshot!($input, ExactDateTime::from_ymd_hms(2023, 1, 1, 0, 0, 0));
    };
}


#[cfg(not(target_family = "wasm"))]
#[test]
fn run_tests(){
    glob!("bloks/*.tb", |path| {
        let input = fs::read_to_string(path).unwrap();
        assert_compile_snapshot!(&input);
    });
}

#[cfg(target_family = "wasm")]
#[test]
fn run_tests(){
    glob!("wsm-bloks/*.tb", |path| {
        let input = fs::read_to_string(path).unwrap();
        assert_compile_snapshot!(&input);
    });
}