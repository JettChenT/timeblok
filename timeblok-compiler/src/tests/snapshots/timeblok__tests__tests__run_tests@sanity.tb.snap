---
source: timeblok-compiler/src/tests/tests.rs
expression: "compile_deterministic(&input,\n        ExactDateTime::from_ymd_hms(2023, 1, 1, 0, 0, 0).clone()).unwrap()"
input_file: timeblok-compiler/src/tests/bloks/sanity.tb
---
BEGIN:VCALENDAR
VERSION:2.0
PRODID:ICALENDAR-RS
CALSCALE:GREGORIAN
BEGIN:VEVENT
DTEND:20230404T023000Z
DTSTAMP:20221231T160000Z
DTSTART:20230404T020000Z
SUMMARY:wake up and eat breakfast
UID:c8e1eb22-a627-3daa-884c-2205cf78d075
END:VEVENT
BEGIN:VEVENT
DTEND:20230404T033000Z
DTSTAMP:20221231T160000Z
DTSTART:20230404T030000Z
SUMMARY:go to work
UID:9311744c-3746-3502-84c9-d06e8b5ea2d6
END:VEVENT
END:VCALENDAR

